pkg/autoscaler/autoscaler.go

```diff
@@ -279,6 +281,9 @@ var _ = Describe("Autoscaler should", framework.LabelAutoscaler, Serial, func()
 			}, framework.WaitLong, pollingInterval).Should(BeTrue())
 		})
 
+		// Machines required for test: 4
+		// Reason: Needs to scale down to minReplicas = 1. Scales 1 -> 2 -> 1. Uses 2 machineSets => 4 machines.
+		// This test copies reproduction steps for a a bug. I have not found a reason why it uses two machineSets. Should be moved to periodic.
```

Comment:
we could probably reduce this to a single machineset. it seems like the main thing we are testing here is that the deletion taint is added to the machine properly.

Reply:
Wasn't sure if there was any special interaction with two machine sets. If you feel it will still test what it should, then I am happy with changing it too. Changed it to use single machineset.